k8s/upgrades/0.9.0-1.0.0/cstor_volume_upgrade.sh

```diff
@@ -0,0 +1,208 @@
+# STEP: Get Persistent Volume (PV) name as argument            #
+#                                                              #
+# NOTES: Obtain the pv to upgrade via "kubectl get pv"         #
+################################################################
+target_upgrade_version="v1.0.x-ci"
+current_version="0.9.0"
+
+function error_msg() {
+    echo "Failed to upgrade volume $pv in namespace $ns. Please make sure that volume upgrade should be successful before moving to application checks"
+}
+
+function usage() {
+    echo
+    echo "Usage:"
+    echo
+    echo "$0 <pv-name> <openebs-namespace>"
+    echo
+    echo "  <pv-name> Get the PV name using: kubectl get pv"
+    echo "  <openebs-namespace> Get the namespace where openebs"
+    echo "  pods are installed"
+    exit 1
+}
+
+if [ "$#" -ne 2 ]; then
+    usage
+fi
+
+pv=$1
+ns=$2
+
+# Check if pv exists
+kubectl get pv $pv &>/dev/null;check_pv=$?
+if [ $check_pv -ne 0 ]; then
+    echo "$pv not found"; error_msg; exit 1;
+fi
+
+# Check if CASType is cstor
+cas_type=`kubectl get pv $pv -o jsonpath="{.metadata.annotations.openebs\.io/cas-type}"`
+if [ $cas_type != "cstor" ]; then
+    echo "Cstor volume not found";exit 1;
+elif [ $cas_type == "cstor" ]; then
+    echo "$pv is a cstor volume"
+else
+    echo "Volume is neither cstor or cstor"; exit 1;
+fi
+
+sc_ns=`kubectl get pv $pv -o jsonpath="{.spec.claimRef.namespace}"`
+sc_name=`kubectl get pv $pv -o jsonpath="{.spec.storageClassName}"`
+sc_res_ver=`kubectl get sc $sc_name -n $sc_ns -o jsonpath="{.metadata.resourceVersion}"`
+pvc_name=`kubectl get pv $pv -o jsonpath="{.spec.claimRef.name}"`
+pvc_namespace=`kubectl get pv $pv -o jsonpath="{.spec.claimRef.namespace}"`
+#################################################################
+# STEP: Generate deploy, replicaset and container names from PV #
+#                                                               #
+# NOTES: Ex: If PV="pvc-cec8e86d-0bcc-11e8-be1c-000c298ff5fc",  #
+#                                                               #
+# c-dep: pvc-cec8e86d-0bcc-11e8-be1c-000c298ff5fc-target        #
+#################################################################
+
+c_dep=$(kubectl get deploy -n $ns \
+        -l openebs.io/persistent-volume=$pv,openebs.io/target=cstor-target \
+        -o jsonpath="{.items[*].metadata.name}")
+c_svc=$(kubectl get svc -n $ns \
+        -l openebs.io/persistent-volume=$pv,openebs.io/target-service=cstor-target-svc \
+        -o jsonpath="{.items[*].metadata.name}")
+c_vol=$(kubectl get cstorvolumes \
+        -l openebs.io/persistent-volume=$pv -n $ns \
+        -o jsonpath="{.items[*].metadata.name}")
+c_replicas=$(kubectl get cvr -n $ns \
+             -l openebs.io/persistent-volume=$pv \
+             -o jsonpath="{range .items[*]}{@.metadata.name};{end}" | tr ";" "\n")
+
+# Fetch the older target and replica - ReplicaSet objects which need to be
+# deleted before upgrading. If not deleted, the new pods will be stuck in
```

Comment:
do we delete before upgrading or after upgrading?

Reply:
Updated the comment